crates/core/src/polynomial/arith_circuit.rs

```diff
@@ -536,6 +531,32 @@ fn apply_binary_op<F: Field, P: PackedField<Scalar: ExtensionField<F>>>(
 	}
 }
 
+fn alloc_scratch_space<T, U, F>(size: usize, callback: F) -> U
+where
+	F: FnOnce(&mut [MaybeUninit<T>]) -> U,
+{
+	#[cfg(miri)]
+	{
+		use std::mem;
+		// We don't want to deal with running destructors.
+		assert!(!mem::needs_drop::<T>());
+		let mut scratch_space = Vec::<T>::with_capacity(size);
+		unsafe {
+			scratch_space.set_len(size);
+			let out =
+				callback(mem::transmute::<&mut [T], &mut [MaybeUninit<T>]>(scratch_space.as_mut()));
```

Comment:
Looks like this is the same as https://doc.rust-lang.org/std/vec/struct.Vec.html#method.spare_capacity_mut

Reply:
Bulls eye! I haven't thought about this...

I am going to risk it and apply the "Graphite-suggested fix"!

Reply:
The AI lied here. You need to move `set_len` to _after_ this line with `spare_capacity_mut()`.

Reply:
Maybe 2026 is going to be the year of AI vibe coding...

Reply:
Wait, why would I need `set_len` at all? We are not interested in the vec logic, only in it's backing storage/capacity...

Reply:
Right, good point. Removing `set_len` is enough then.